### ISSUE TITLE
Missing ('z') dataset in CloudyData_UVB=HM2012_high_density.h5 input data file

### ISSUE THREAD

**Issue by mrabenanahary** [grackle-project/grackle_data_files#6]
Hi everyone,

I'm a PhD student, I'm new with Grackle, and i'm currently trying to add it on my PDR code to add chemistry and H2 cooling.

Maybe I got something wrong (and I'm not familiar with HDF5), but I encountered an issue when trying to execute my code with the new input data file `CloudyData_UVB=HM2012_high_density.h5`, instead of the default `CloudyData_UVB=HM2012.h5` (I really need to deal with higher densities physics). 

For my code, I need to enable dust chemistry and metal cooling.

More precisely, there is no issue during execution when I set `grackle_data%grackle_data_file = CloudyData_UVB=HM2012.h5`. But when I set `grackle_data%grackle_data_file = CloudyData_UVB=HM2012_high_density.h5`, I get that HDF5 can't open redshift dataset ('z'), with the following shell output (displayed when executing `iresult = initialize_chemistry_data(my_units)`) :

```
The Grackle Version 3.2.dev2
Git Branch   main
Git Revision f9c828a4155747071c41407269625a867541c634

Initializing grackle data.
Dust chemistry enabled, setting photoelectric_heating to 2.
Dust chemistry enabled, setting dust_recombination_cooling to 1.
Dust chemistry enabled, setting h2_on_dust to 1.
Initializing Cloudy cooling: Metals.
cloudy_table_file: ../../input/CloudyData_UVB=HM2012_high_density.h5.
Cloudy cooling grid rank: 3.
Cloudy cooling grid dimensions: 42 26 161.
Parameter1: -10 to 10 (42 steps).
Parameter2: 0 to 15 (26 steps).
Temperature: 1 to 9 (161 steps).
Reading Cloudy Cooling dataset.
Reading Cloudy Heating dataset.
Initializing UV background.
Reading UV background data from ../../input/CloudyData_UVB=HM2012_high_density.h5.
HDF5-DIAG: Error detected in HDF5 (1.10.4) thread 140689315553280:
  #000: ../../../src/H5Ddeprec.c line 195 in H5Dopen1(): unable to open dataset
    major: Dataset
    minor: Can't open object
  #001: ../../../src/H5Dint.c line 1205 in H5D__open_name(): not found
    major: Dataset
    minor: Object not found
  #002: ../../../src/H5Gloc.c line 422 in H5G_loc_find(): can't find object
    major: Symbol table
    minor: Object not found
  #003: ../../../src/H5Gtraverse.c line 851 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #004: ../../../src/H5Gtraverse.c line 627 in H5G__traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #005: ../../../src/H5Gloc.c line 378 in H5G__loc_find_cb(): object 'z' doesn't exist
    major: Symbol table
    minor: Object not found
Can't open redshift dataset ('z') in ../../input/CloudyData_UVB=HM2012_high_density.h5.
Error in initialize_UVbackground_data.
Error in _initialize_chemistry_data.
Error in initialize_chemistry_data.
```

I tried to look at the content of the two `.h5` files with h5web Jupyter extension, but as you can see it below, `CloudyData_UVB=HM2012_high_density.h5` really misses a 'z' entry under the 'UVBrates' group, while it is clearly here for `CloudyData_UVB=HM2012.h5` :

![Capture d’écran de 2022-08-06 13-10-35](https://user-images.githubusercontent.com/63724500/183247192-b5b58c43-ae98-4fe5-ae19-d8f3c55b08ed.png)
![Capture d’écran de 2022-08-06 13-10-14](https://user-images.githubusercontent.com/63724500/183247196-bdf6f1f6-aa88-4a3f-8ccd-76dcae86c702.png)

How may I deal with this issue, please ?

Any assistance would be appreciated. Thank you !

**Comment by brittonsmith** [comment]
Hi there!

Apologies for taking so long to get back to you on this issue. I've compared some of the rates between `CloudyData_UVB=HM2012_high_density.h5` and `CloudyData_UVB=HM2012.h5` and it looks like they are exactly the same. I've provided a short python script below that will copy the redshift data from one file and add it to the other. Can you run this and test out the modified file and let me know if everything works? If so, I will issue a pull request with the modified file.

```
import h5py

fh1 = h5py.File("CloudyData_UVB=HM2012.h5", mode="r")
z_data = fh1["UVBRates/z"][()]
fh1.close()

fh2 = h5py.File("CloudyData_UVB=HM2012_high_density.h5", mode="a")
fh2["UVBRates"].create_dataset("z", data=z_data)
fh2.close()
```

**Comment by mrabenanahary** [comment]
Hi,

Thank you for your reply ! I successfully appendend the redshift `UVBRates/z` dataset to `CloudyData_UVB=HM2012_high_density.h5`. I only ran it on the `fortran_example` example script. 
Even if I get different values compared to when I'm using
`CloudyData_UVB=HM2012.h5`, everything else seems to work fine. I'm still working on testing it on my code and with testing physical cases, to see if I get the expected results !

Output with `CloudyData_UVB=HM2012.h5`:
 ```
primordial_chemistry:           3
 metal_cooling:           1
 Cooling time =   -217289250719936.25      s.
 Temperature =    2550.7903685639185      K.
 Pressure =    3.2854922648446871E-013 dyne/cm^2.
 Gamma =    1.6665433133050476     
 Dust temperature =    40.832709368784194      K.
```

Output with `CloudyData_UVB=HM2012_high_density.h5`:
```
 primordial_chemistry:           3
 metal_cooling:           1
 Cooling time =    68842583497299.016      s.
 Temperature =    1407.5257168842161      K.
 Pressure =    1.7641570805682165E-013 dyne/cm^2.
 Gamma =    1.6665739181586814     
 Dust temperature =    40.861189734683478      K.
```

Thank you for your help !

**Comment by brittonsmith** [comment]
@mrabenanahary, how are things going with the corrected table? If you're happy with how things are looking, would you be interested in issuing a pull request to update this table in the main repository?

**Comment by brittonsmith** [comment]
Resolved in PR #9.
